rdt-app/src/main/java/io/ona/rdt_app/fragment/RDTJsonFormFragment.java

```diff
@@ -58,13 +64,34 @@ public void onClick(View v) {
                 Object isSubmit = v.getTag(R.id.submit);
                 if (isSubmit != null && Boolean.valueOf(isSubmit.toString())) {
                     save(false);
+                    checkIfContinuingToRdt();
+                    // Check if we need to continue to RDT
                 } else {
                     next();
                 }
             }
         });
     }
 
+    public void checkIfContinuingToRdt() {
```

Comment:
Could you move this logic to `PatientRegisterFragmentInteractor`.

This is to avoid having complex logic in the views.

Also, the `saveForm` method there receives a jsonform json object, and we do not need to do the extra first steps of processing here.

Reply:
Also, this method should just check if the rdt test should continue or not.

The launching of the form should be done by the calling method.

Reply:
Cool, will have each method do one thing